server/pulp/server/webservices/views/content.py

```diff
@@ -1,13 +1,378 @@
+        """
+        Delete a specified orphan.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param content_type: content type of the requested orphan
+        :type  content_type: string
+        :param unit_id: id of the requested orphan
+        :type  unit_id: string
+
+        :raises: OperationPostponed
+        """
+        orphan_manager = factory.content_orphan_manager()
+        unit_info = [{'content_type_id': content_type, 'unit_id': unit_id}]
+        task_tags = [tags.resource_tag(tags.RESOURCE_CONTENT_UNIT_TYPE, 'orphans')]
+        async_task = orphan_manager.delete_orphans_by_id.apply_async((unit_info,), tags=task_tags)
+
+        raise OperationPostponed(async_task)
+
+
+class DeleteOrphansActionView(View):
+    """
+    Deprecated in 2.4, please use the more restful OrphanResource delete instead
+
+    Delete all orphans.
+    """
+
+    @auth_required(authorization.DELETE)
+    def post(self, request):
+        """
+        Pass the appropriate tags and json object to delete_orphans_by_id.
+        Raises an OperationPostponed.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+
+        :raises: OperationPostponed
+        """
+        orphan_manager = factory.content_orphan_manager()
+        all_orphans = request.body_as_json
+        task_tags = [tags.action_tag('delete_orphans'),
+                     tags.resource_tag(tags.RESOURCE_CONTENT_UNIT_TYPE, 'orphans')]
+        async_task = orphan_manager.delete_orphans_by_id.apply_async([all_orphans], tags=task_tags)
+
+        raise OperationPostponed(async_task)
+
+
+class CatalogResourceView(View):
+    """
+    Views for the catalog by source_id.
+    """
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, source_id):
+        """
+        Delete entries from the catlog by content source id
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param source_id: id of source whose content should be deleted
+        :type  source_id: string
+
+        :return: response containing a dict containing the number if items deleted
+        :rtype : django.http.HttpResponse
+        """
+        manager = factory.content_catalog_manager()
+        purged = manager.purge(source_id)
+        deleted_info = dict(deleted=purged)
+        return generate_json_response(deleted_info)
+
+
+class ContentTypeResourceView(View):
+    """
+    View for a single content type.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, type_id):
+        """
+        Return a response containing a dict with info about a content type.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param type_id: type of content unit
+        :type  type_id: str
```

Comment:
On 01/28/2015 08:31 AM, bmbouter wrote:
> s/str/string/ 'str' isn't a valid type.

str is a valid type:

https://docs.python.org/2/library/functions.html#str